controllers/clustergroupupgrade_controller.go

```diff
@@ -327,9 +327,23 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 						nextReconcile = ctrl.Result{RequeueAfter: requeueAfter}
 					}
 
-					// Check for batch timeout.
-					batchTimeout := time.Duration(clusterGroupUpgrade.Spec.RemediationStrategy.Timeout/len(clusterGroupUpgrade.Status.RemediationPlan)) * time.Minute
-					if !clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.IsZero() && time.Since(clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.Time) > batchTimeout {
+					// The remaining time will be the total timeout subtract the elapsed time spent on both the batch and cgu
+					// It is important to include the current batch here so that we don't let the entire timeout be consumed by a single batch that gets stuck
+					remainingTime := float64(clusterGroupUpgrade.Spec.RemediationStrategy.Timeout) - clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.Time.Sub(clusterGroupUpgrade.Status.Status.StartedAt.Time).Seconds()
```

Comment:
is the clusterGroupUpgrade.Spec.RemediationStrategy.Timeout always in sec ? better to create time.Duration and then subtract the seconds

Reply:
As far as I can tell this is always in seconds, but it is just an integer. It is also treated as seconds elsewhere. I would still have to assume this is seconds to make a duration no?